components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -0,0 +1,1299 @@
+        wet_aero_.int_aero_mmr[m][a] =
+            get_field_out(int_mmr_field_name).get_view<Real **>();
+        dry_aero_.int_aero_mmr[m][a] = buffer_.dry_int_aero_mmr[m][a];
+      }
+
+      // (cloudborne) aerosol tracers of interest: mass (q) mixing ratios
+      const char *cld_mmr_field_name =
+          mam_coupling::cld_aero_mmr_field_name(m, a);
+      if(strlen(cld_mmr_field_name) > 0) {
+        wet_aero_.cld_aero_mmr[m][a] =
+            get_field_out(cld_mmr_field_name).get_view<Real **>();
+        dry_aero_.cld_aero_mmr[m][a] = buffer_.dry_cld_aero_mmr[m][a];
+      }
+    }
+  }
+  for(int g = 0; g < mam_coupling::num_aero_gases(); ++g) {
+    const char *gas_mmr_field_name = mam_coupling::gas_mmr_field_name(g);
+    wet_aero_.gas_mmr[g] =
+        get_field_out(gas_mmr_field_name).get_view<Real **>();
+    dry_aero_.gas_mmr[g] = buffer_.dry_gas_mmr[g];
+  }
+
+  // hetrozenous freezing outputs
+  hetfrz_immersion_nucleation_tend_ =
+      get_field_out("hetfrz_immersion_nucleation_tend").get_view<Real **>();
+  hetfrz_contact_nucleation_tend_ =
+      get_field_out("hetfrz_contact_nucleation_tend").get_view<Real **>();
+  hetfrz_depostion_nucleation_tend_ =
+      get_field_out("hetfrz_depostion_nucleation_tend").get_view<Real **>();
+
+  //---------------------------------------------------------------------------------
+  // Allocate memory for the class members
+  // (Kokkos::resize only works on host to allocates memory)
+  //---------------------------------------------------------------------------------
+
+  Kokkos::resize(rho_, ncol_, nlev_);
+  Kokkos::resize(w0_, ncol_, nlev_);
+  Kokkos::resize(tke_, ncol_, nlev_ + 1);
+  Kokkos::resize(wsub_, ncol_, nlev_);
+  Kokkos::resize(wsubice_, ncol_, nlev_);
+  Kokkos::resize(wsig_, ncol_, nlev_);
+  Kokkos::resize(w2_, ncol_, nlev_);
+  Kokkos::resize(cloud_frac_, ncol_, nlev_);
+  Kokkos::resize(cloud_frac_prev_, ncol_, nlev_);
+  Kokkos::resize(aitken_dry_dia_, ncol_, nlev_);
+  Kokkos::resize(rpdel_, ncol_, nlev_);
+
+  //---------------------------------------------------------------------------------
+  // Diagnotics variables from the ice nucleation scheme
+  //---------------------------------------------------------------------------------
+
+  // number conc of ice nuclei due to heterogeneous freezing [1/m3]
+  Kokkos::resize(nihf_, ncol_, nlev_);
+
+  // number conc of ice nuclei due to immersionfreezing (hetero nuc) [1/m3]
+  Kokkos::resize(niim_, ncol_, nlev_);
+
+  // number conc of ice nuclei due to deposition nucleation (hetero nuc)[1/m3]
+  Kokkos::resize(nidep_, ncol_, nlev_);
+
+  // number conc of ice nuclei due to meyers deposition [1/m3]
+  Kokkos::resize(nimey_, ncol_, nlev_);
+
+  // number of activated aerosol for ice nucleation(homogeneous frz only)[#/kg]
+  Kokkos::resize(naai_hom_, ncol_, nlev_);
+
+  //---------------------------------------------------------------------------------
+  // Diagnotics variables from the droplet activation scheme
```

Comment:
It would be nice to have the CCN diagnostic output fields (e.g., CCN@0.1%, 0.2%, and 0.5%). Perhaps this can be addressed in a future PR.